packages/bappo-components/src/primitives/Checkbox/Checkbox.native/Checkbox.tsx

```diff
@@ -7,85 +7,105 @@ import {
 
-export default function Checkbox({
-  accessibilityLabel,
-  checked = false,
-  colorChecked = '#FF7800',
-  colorUnchecked = '#B0ADAB',
-  size = 'medium',
-  disabled = false,
-  style,
-  onBlur,
-  onFocus,
-  onValueChange,
-  testID,
-}: CheckboxProps) {
-  const sizes = {
-    small: 16,
-    medium: 24,
-    large: 32,
-  };
-
-  const _onBlur = (_event: NativeSyntheticEvent<TargetedEvent>) => {
-    onBlur?.({
-      nativeEvent: {
-        checked,
-      },
-    });
-  };
-
-  const _onFocus = (_event: NativeSyntheticEvent<TargetedEvent>) => {
-    onFocus?.({
-      nativeEvent: {
-        checked,
-      },
-    });
-  };
-
-  const _onKeyPress = (event: React.KeyboardEvent<TouchableOpacity>) => {
-    const ENTER = 13;
-    const SPACE = 32;
-
-    if (event.which === ENTER || event.which === SPACE) {
-      event.preventDefault();
-      event.stopPropagation();
-      _toggle();
-    }
-  };
-
-  const _toggle = () => !disabled && onValueChange?.(!checked);
-
-  const containerprops = {
-    accessibilityLabel,
-    onBlur: _onBlur,
-    onPress: _toggle,
-    onFocus: _onFocus,
-    onKeyPress: _onKeyPress,
-    testID,
-    $checked: !!checked,
-    $disabled: disabled,
-  };
-
-  const styleProps = {
-    style,
-  };
-
-  const icon = checked ? 'check-box' : 'check-box-outline-blank';
-  const color = disabled ? 'gray' : checked ? colorChecked : colorUnchecked;
-
-  return (
-    <CheckboxContainer $size={sizes[size]} {...containerprops} {...styleProps}>
-      <CheckIcon
-        name={icon}
-        color={color}
-        size={sizes[size]}
-        $disabled={disabled}
-      />
-    </CheckboxContainer>
-  );
-}
+const Checkbox = React.forwardRef(
+  (
+    {
+      accessibilityLabel,
+      checked = false,
+      colorChecked = '#FF7800',
+      colorUnchecked = '#B0ADAB',
+      size = 'medium',
+      disabled = false,
+      style,
+      onBlur,
+      onFocus,
+      onValueChange,
+      testID,
+    }: CheckboxProps,
+    ref: React.Ref<InputHandle>,
+  ) => {
+    const containerRef = React.useRef<TouchableOpacity>(null);
+
+    React.useImperativeHandle(ref, () => ({
+      focus: () => containerRef.current?.focus(),
+      blur: () => containerRef.current?.blur(),
+    }));
+
+    const sizes = {
+      small: 16,
+      medium: 24,
+      large: 32,
+    };
+
+    const _onBlur = (_event: NativeSyntheticEvent<TargetedEvent>) => {
+      onBlur?.({
+        nativeEvent: {
+          checked,
+        },
+      });
+    };
+
+    const _onFocus = (_event: NativeSyntheticEvent<TargetedEvent>) => {
+      onFocus?.({
+        nativeEvent: {
+          checked,
+        },
+      });
+    };
+
+    const _onKeyPress = (event: React.KeyboardEvent<TouchableOpacity>) => {
+      const ENTER = 13;
+      const SPACE = 32;
+
+      if (event.which === ENTER || event.which === SPACE) {
+        event.preventDefault();
+        event.stopPropagation();
+        _toggle();
+      }
+    };
```

Comment:
Do we need this for native?

Reply:
The `TouchableOpacity` component allows for keyboard input so I kept it to be consistent with web but I also think it is unnecessary and removed it.